src/lib/util/simh_tape_file.cpp

```diff
@@ -0,0 +1,624 @@
+// license:BSD-3-Clause
+// copyright-holders:Mietek Bak
+
+// best read together with SIMH magtape spec (rev 17 Jan 2022)
+// http://simh.trailing-edge.com/docs/simh_magtape.pdf
+
+#include <cassert>
+#include <cstring>
+#include <stdexcept>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/mman.h>
```

Comment:
As already mentioned, mmap() is different on Windows. Either replace with conventional file I/O or propose additions to OSD to support it. Is it really beneficial or necessary given typical memory capacity and "legacy" tape capacities?

Reply:
The `mmap` approach is not used elsewhere in MAME for file format code (its only use is in `src/mame/gaelco/gaelco3d_m.cpp` for some rather idiosyncratic networking code), and smells very bad to me.

Reply:
Why?  To me, the `mmap` approach seems like a natural fit for image files in general, considering their size and typical access patterns.  Loading a ~700MB CD-ROM image file in MAME results in a significant delay.  Tape images can potentially run into multiple GBs.

Reply:
MAME does not load an entire CD-ROM or hard disk image all at once, it just reads and writes the current block.  The startup delay is due to an integrity check.  As has been noted, we don't use mmap() for any other file I/O.  If you want to propose a portable mmap() implementation for MAME's file I/O layer that's a thing you can do in a separate PR, but otherwise this one needs to use MAME's file I/O.